lang/java/reef-experimental/src/main/java/org/apache/reef/experimental/ParquetReader.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.experimental;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.avro.AvroSchemaConverter;
+import org.apache.parquet.format.converter.ParquetMetadataConverter.MetadataFilter;
+import static org.apache.parquet.format.converter.ParquetMetadataConverter.NO_FILTER;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+
+import javax.inject.Inject;
+
+/**
+ * A reader for Parquet files that can serialize data to local disk and return Avro schema and Avro reader.
+ * The intent is not to build a general parquet reader, but to consume data with table-like property.
+ */
+public class ParquetReader {
+  /**
+   * Standard java logger.
+   */
+  private static final Logger LOG = Logger.getLogger(ParquetReader.class.getName());
+  
+  private Path parquetFilePath;
+
+  @Inject
+  public ParquetReader(final Path path) throws IOException {
+    parquetFilePath = path;
+    Schema schema = getAvroSchema();
+    if (schema.getType() != Schema.Type.RECORD) {
+      LOG.log(Level.SEVERE, "ParquetReader only support Avro record type that can be consumed as a table.");
+    }
+    for (Schema.Field f : schema.getFields()) {
+      if (f.schema().getType() == Schema.Type.RECORD) {
+        LOG.log(Level.SEVERE, "ParquetReader doesn't support nested record type for its elements.");
+      }
+    }
+  }
+  
+  /**
+   * Retrieve avro schema from parquet file.
+   * @return avro schema from parquet file.
+   */
+  public Schema getAvroSchema() throws IOException {
+    return getAvroSchema(new Configuration(true), NO_FILTER);
+  }
+  
+  /**
+   * Retrieve avro schema from parquet file.
+   * @param configuration Hadoop configuration.
+   * @param filter Filter for Avro metadata.
+   * @return avro schema from parquet file.
+   */
+  public Schema getAvroSchema(final Configuration configuration, final MetadataFilter filter) throws IOException {
```

Comment:
Does this need to be `public`? If so, please document the exception with `@throws`